src/utilities/plugin-utils.ts

```diff
@@ -51,35 +55,69 @@ export const loadPluginScripts = (activity: Activity) => {
   });
 };
 
-export const initializePlugin = (embeddable: IEmbeddablePlugin, wrappedEmbeddable: Embeddable | undefined, embeddableContainer: HTMLElement, wrappedEmbeddableContainer: HTMLElement | undefined) => {
+export interface IEmbeddablePluginContext {
+  LARA: LaraGlobalType;
+  embeddable: IEmbeddablePlugin;
+  embeddableContainer: HTMLElement;
+  wrappedEmbeddable?: Embeddable;
+  wrappedEmbeddableContainer?: HTMLElement;
+  sendCustomMessage?: (message: ICustomMessage) => void;
+}
+export type IPartialEmbeddablePluginContext = Partial<IEmbeddablePluginContext>;
+
+export const validateEmbeddablePluginContextForPlugin =
+              (context: IPartialEmbeddablePluginContext): IEmbeddablePluginContext | undefined => {
+  const { LARA, embeddable, embeddableContainer, ...others } = context;
+  return LARA && embeddable && embeddableContainer
+          ? { LARA, embeddable, embeddableContainer, ...others }
+          : undefined;
+};
+
+export const validateEmbeddablePluginContextForWrappedEmbeddable =
+              (context: IPartialEmbeddablePluginContext): IEmbeddablePluginContext | undefined => {
+  const { wrappedEmbeddable, wrappedEmbeddableContainer } = context;
+  const validated = validateEmbeddablePluginContextForPlugin(context);
+  return validated && wrappedEmbeddable && wrappedEmbeddableContainer ? validated : undefined;
+};
+
+export const initializePlugin = (context: IEmbeddablePluginContext) => {
+  const { LARA, embeddable, embeddableContainer,
+          wrappedEmbeddable, wrappedEmbeddableContainer, sendCustomMessage } = context;
   // TODO: will need to change search as we implement other plugin types
   const plugin = Plugins.find(p => p.type === "TeacherEdition");
+  if (!plugin) return;
```

Comment:
Ah, this ties to my earlier comment about the plugin scripts not being loaded.  We need to keep an eye on cases where we might come into here before scripts are loaded.

Reply:
I agree that it's out of scope for this PR. The answer, however, comes from the fact that `initializePlugin()` is called from `useEffect()` hooks that have a bunch of dependencies, so how many times it gets called will depend on the number of times that those hooks' dependencies change and how that relates to the time it takes for the plugin to load. So there could be bugs related to when `initializePlugin()` gets called relative to when the script loads and there also could be issues related to calling `initializePlugin()` too often (or perhaps not enough, although that seems less likely).

Reply:
Yes this does seem like it will cause problems. 

In LARA authoring this is handled here:
https://github.com/concord-consortium/lara/blob/f7566989290dd7100ee764ab93103e72e503eb36/app/views/plugins/_author.haml#L72-L101
A handler is added to the script.onload event, before adding the script element to the page. And initPlugin is called in the handler.

However that is only how it is done during authoring, which only loads one interactive's plugins at a time and reloads the page afterwards.

At runtime it is handled here: 
https://github.com/concord-consortium/lara/blob/f7566989290dd7100ee764ab93103e72e503eb36/app/views/plugins/_show.haml#L2-L7
followed by this
https://github.com/concord-consortium/lara/blob/f7566989290dd7100ee764ab93103e72e503eb36/app/views/plugins/_show.haml#L148-L178
Due to browsers synchronous loading of javascript, that guarantees the initInteractive won't be called until the plugin is loaded.

I think the AP will need to assign unique plugin labels to the plugins referenced by the activity. And then need to resolve a promise when a plugin with that label is registered. Then the code here can wait for that promise to be resolved before calling initPlugin.

Reply:
It seems clear there's a larger body of work here, which seems better left to a subsequent PR.